### ISSUE TITLE
Catch sometimes leads to compile error

### ISSUE THREAD

**Issue by markspanbroek** [arnetheduck/nim-result#21]
The following code leads to a compile error:

```nim
import std/strutils
import pkg/result/../results

let error = 42 # a symbol "error" needs to be in scope to trigger the issue
discard parseInt("foo").catch.error # Error: the field 'v' is not accessible.
```

The `catch` template uses the `ok` and `err` templates, which in combination with the ambiguity of `error` in this code, leads to a situation where Nim tries to evaluate the code from the `ok` template in a scope where `v` is not visible.

It can be solved by making `ok` and `err` procs instead of templates.

**Comment by arnetheduck** [comment]
cc @zah this is the nim early symbol matching kicking in, right? Is there anything we can do in result (ie detect it and print an informative error at least?)

**Comment by arnetheduck** [comment]
"fixed" in #26
